src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea: () => number;
+}
+
+function areaRound(area: number): number {
+  return Math.floor(area * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: string = `triangle`,
+  ) {
+    const sides = [a, b, c].sort((side1, side2) => side1 - side2);
+    const isValidSides = sides[2] < sides[0] + sides[1];
+
+    if (a <= 0 || b <= 0 || c <= 0 || !isValidSides) {
+      throw new Error('any side must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
-export class Triangle implements Figure {}
+    return areaRound(area);
```

Comment:
According to the checklist, the `getArea` method should not accept parameters as all necessary values are available as object fields. Remove the parameter from the method signature.